DirectProgramming/DPC++FPGA/PlatformTests/board_test/src/board_test.hpp

```diff
@@ -0,0 +1,1852 @@
+    }
+
+    // **** Write to device global memory **** //
+
+    // Submit copy operation (explicit copy from host to device)
+    auto h2d_copy_e = q.submit([&](handler &h) {
+      // Range of buffer that needs to accessed = i bytes (chars)
+      // Device buffer is accessed at dev_offset
+      // Using +3 offset on aligned device pointer ensures that DMA is never
+      // used (because the host ptr is not aligned)
+      accessor mem(dev_buf, h, i, dev_offset);
+      // Writing from host memory to device buffer
+      h.copy(host_in_buf, mem);
+    });
+    // Wait for copy to complete
+    h2d_copy_e.wait();
+
+    // **** Read from device global memory **** //
+
+    // Submit copy operation (explicit copy from device to host)
+    auto d2h_copy_e = q.submit([&](handler &h) {
+      // Range of buffer that needs to accessed = i bytes (chars)
+      // Device buffer is accessed at dev_offset
+      accessor mem(dev_buf, h, i, dev_offset);
+      // Reading from device buffer into host memory
+      h.copy(mem, host_rd_buf);
+    });
+    // Wait for copy to complete
+    d2h_copy_e.wait();
+
+// Verify values read back match the input
+#ifdef DEBUG
+    std::cout << host_rd_buf[0] << " , " << host_in_buf[0] << "\n";
+#endif
+
+    if (memcmp(host_in_buf, host_rd_buf, i) != 0) {
+      std::cout << i << " bytes read/write FAILED!\n";
+      for (size_t m = 0; m < i; m++) {
+        if (host_in_buf[m] != host_rd_buf[m]) {
+          std::cout << "char #" << m
+                    << " , host input buffer = " << host_in_buf[m]
+                    << " , host read back buffer = " << host_rd_buf[m] << "\n";
+        }
+      }
+      assert(0);
+    }
+
+    // make sure bounds on both ends of buffer are ok
+    for (signed int k = (-1 * kHostRdOffset); k < kHostRdOffset; k++) {
+      if (k < 0)
+        assert(host_rd_buf[k] == invalid_read_back);
+      else
+        assert(host_rd_buf[i + k] == invalid_read_back);
+    }
+
+  }  // End of for loop for writing/reading all bytes
+
+  // Free allocated host memory
+  free(host_in_buf_ptr);
+  free(host_rd_buf_ptr);
+
+  // Unaligned transfers completed successfully, test passed
+  h2d_rw_test = true;
+  return 0;
+}
+
+//////////////////////////////////////
+// **** KernelClkFreq function **** //
+//////////////////////////////////////
+
+// Inputs:
+// queue &q - queue to submit operation
+// bool report_chk - control if Quartus compiled frequency should be read
+// Returns:
+// 0 is test passes, 1 if test fails
+
+// The function does the following tasks:
+// 1. Launches an no-operation NDRange kernel with global size 128 MB
+// 2. Measures time taken for the above NDRange kernel using Sycl event
+// profiling information
+// 3. Obtain kernel clock frequency based on time take for 128 Mglobal
+// operations (NDRange)
+// 4. If the <report_chk> is true, compare the above hardware frequency with
+// Quartus compiled frequency NOTE: This needs the reports or
```

Comment:
```suggestion
// Quartus compiled frequency 
// NOTE: This needs the reports or
```

Reply:
This NOTE is no longer needed in the code and is removed.